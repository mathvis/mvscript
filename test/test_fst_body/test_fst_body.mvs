func a() {
	let c = 0;
}
