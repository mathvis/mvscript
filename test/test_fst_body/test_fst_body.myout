Decl (FunctionDef (FunctionIdentifier "a") [] void (Just (Block (FunctionBlock void) [Decl (Variable (VarIdentifier "c") (Just int) (Just (Type (Int 0))))])))

--- STATE ---

Function symbol table:
	Function name: a
		Return type: void
		Arguments: 
		Has body: True
