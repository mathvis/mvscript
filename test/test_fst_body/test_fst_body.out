(Decl (FunctionDef (FunctionIdentifier "a") [] VoidT (Just (Block (FunctionBlock VoidT) [Decl (Variable (VarIdentifier "c") (Just IntT) (Just (Type (Int 0))))]))),ParserState {config = Configuration {debug = True, collapseOperations = True, collapseControlFlow = True, colors = True}, st = fromList [("c",VariableData {variableType = Just IntT, inScope = False, isInitialized = True, isConstant = False})], fst = fromList [("a",FunctionData {returnType = VoidT, arguments = [], hasBody = True})], context = Nothing})
